pkg/config/config.go

```diff
@@ -20,17 +20,17 @@ type Config struct {
 	ImageResourceGroup string
 	ImageResourceName  string
 
+	Certificates CertificateConfig
```

Comment:
@pweil- overall the structure LGTM.

question: I am assuming `Certificates` or in fact the whole `Config` will need to be stored. Shouldn't it be a field of: https://github.com/openshift/openshift-azure/blob/master/pkg/api/types.go#L20 as well for storage?

Reply:
@amanohar yes, definitely.  If this is good, next steps on this PR are:

* nest the rest of the certs
* do all the refactor for nesting

Then I'll make a second PR to:

* make reference to Config in the internal types
* do all the refactor for moved config

How about the kubeconfigs, should they be nested too so they can be encrypted for storage or are they good as-is?  Probably as a follow up, trying to keep these small for easier review.

Reply:
kubeconfigs will be encrypted & stored as well but won't be embedded inside the internal data model object for easy retrieval, storage for operations tools and also update by the RP/service so that it doesn't impact the core resource definition and/or operations happening on the resource. 

This for general schema design hygiene and following good DB practices.

Reply:
Hmm, they are currently on the Config object so they will become part of the internal model when I make reference to it.  We can discuss later.  I'll open an issue since it's orthogonal to this refactor https://github.com/openshift/openshift-azure/issues/157